core/services/ocr2/delegate.go

```diff
@@ -147,9 +137,23 @@ func (d Delegate) ServicesForSpec(jobSpec job.Job) ([]job.ServiceCtx, error) {
 	jobSpec.PipelineSpec.JobID = jobSpec.ID
 
 	var pluginOracle plugins.OraclePlugin
+	var ocr2Provider types.OCR2Base
 	switch spec.PluginType {
 	case job.Median:
-		pluginOracle, err = median.NewMedian(jobSpec, ocr2Provider, d.pipelineRunner, runResults, lggr, ocrLogger)
+		medianProvider, err2 := relayer.NewMedianProvider(types.PluginArgs{
+			ConfigWatcherArgs: types.ConfigWatcherArgs{
+				ExternalJobID: jobSpec.ExternalJobID,
+				JobID:         spec.ID,
+				ContractID:    spec.ContractID,
+				RelayConfig:   spec.RelayConfig.Bytes(),
+			},
+			TransmitterID: spec.TransmitterID.String,
+		})
+		if err2 != nil {
+			return nil, err
+		}
+		ocr2Provider = medianProvider
+		pluginOracle, err = median.NewMedian(jobSpec, medianProvider, d.pipelineRunner, runResults, lggr, ocrLogger)
```

Comment:
```suggestion
			return nil, err2
		}
		ocr2Provider = medianProvider
		pluginOracle, err2 = median.NewMedian(jobSpec, medianProvider, d.pipelineRunner, runResults, lggr, ocrLogger)
		if err2 != nil {
			return nil, err2
		}
```

Reply:
Oh, there is an error check after the select - but 153 should still be `err2`, yeah?

Reply:
I didn't notice that until afterwards, which maybe speaks to the readability, but regardess, line 153 is returning `err` after checking `err2`, so that part still applies.